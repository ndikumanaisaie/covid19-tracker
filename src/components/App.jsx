import React, { useEffect } from 'react';
import { Routes, Route } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getCovidData } from '../redux/covidData/covidDataSlice';
import CountryDetails from '../pages/CountryDetails';
import Regions from '../pages/Regions';

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getCovidData());
  }, [dispatch]);

  const { covidData } = useSelector((state) => ({ ...state.covidData }));
  console.log('covidData', covidData);
  return (
    <div>
      <div className="main-container">
        <Routes>
          <Route path="/" element={<Regions />} />
          <Route
            path="/country-details/:id"
            element={<CountryDetails />}
          />
        </Routes>
      </div>
      <h1>
        Hallo React
      </h1>
      {
        covidData.map((data) => (
          <h2 key={data.id}>
            { data.id }
          </h2>
        ))
      }
    </div>
  );
};

export default App;
