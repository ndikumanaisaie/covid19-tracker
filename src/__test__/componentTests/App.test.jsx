import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import App from '../../components/App';
import store from '../../redux/configureStore';

describe('App tests', () => {
  it('it renders', () => {
    const app = renderer
      .create(
        <MemoryRouter>
          <Provider store={store}>
            <App />
          </Provider>
        </MemoryRouter>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });
});
