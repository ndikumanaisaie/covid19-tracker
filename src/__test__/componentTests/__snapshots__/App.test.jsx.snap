// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests it renders 1`] = `
<div
  className="main-container"
>
  <div
    className="container-fluid"
  >
    <select
      aria-label="Select Continent"
      className="continent-select form-select"
      name="category"
      onChange={[Function]}
    >
      <option>
        Continents
      </option>
      <option
        value="North America"
      >
        North America
      </option>
      <option
        value="South America"
      >
        South America
      </option>
      <option
        value="Asia"
      >
        Asia
      </option>
      <option
        value="Africa"
      >
        Africa
      </option>
      <option
        value="Europe"
      >
        Europe
      </option>
      <option
        value="Australia/Oceania"
      >
        Australia/Oceania
      </option>
    </select>
    <div
      className="continent-card card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title h5"
        >
          Africa
        </div>
        <div
          className="mb-2 card-subtitle h6"
        >
          0
           
          Cases
        </div>
      </div>
    </div>
    <div
      className="case"
    >
      <p>
        NUMBER OF CASES BY COUNTRY
      </p>
    </div>
    <div
      className="card-container"
    />
  </div>
</div>
`;
